aws-greengrass-testing-features/aws-greengrass-testing-features-api/src/main/java/com/aws/greengrass/testing/features/SystemSteps.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.testing.features;
+
+import io.cucumber.guice.ScenarioScoped;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import oshi.SystemInfo;
+import oshi.hardware.CentralProcessor;
+import oshi.hardware.GlobalMemory;
+import oshi.hardware.HardwareAbstractionLayer;
+
+import java.util.ArrayList;
+
+
+@ScenarioScoped
+public class SystemSteps {
+    private static final Logger LOGGER = LogManager.getLogger(SystemSteps.class);
+    private ArrayList<Double> ramList = new ArrayList<Double>();
+    private ArrayList<Double> cpuList = new ArrayList<Double>();
+    private SystemInfo si = new SystemInfo();
+    private HardwareAbstractionLayer hal = si.getHardware();
+    private CentralProcessor cpu = hal.getProcessor();
+    private GlobalMemory ram = hal.getMemory();
+
+    /**
+     * Record the current CPU or RAM usage depending on the step.
+     *
+     * @param stat the statistic to record, must be CPU or RAM
+     * @throws IllegalArgumentException when parameter is not CPU or RAM
+     */
+    @When("I record the device's {word} usage statistic")
+    public void recordCpuOrRam(String stat) throws IllegalArgumentException {
+        switch (stat) {
+            case "CPU":
+                double cpuLoad = cpu.getSystemCpuLoad(500) * 100;
```

Comment:
What does `500` signify here?

Reply:
Milliseconds to wait as the time period we check the cpu load over

Reply:
Suggestion: This is good. But, can we have a step that stores the recorded value in scenario context? This will help us test the difference between any two values instead of just the last two. Eg. `I record the device's {word} usage statistic as {word}` and the test would look the following. 

```
    When I record the device's RAM usage statistic as RAM_1
    And I wait 10 seconds
    When I record the device's RAM usage statistic as RAM_2
    And I wait 10 seconds
    When I record the device's RAM usage statistic as RAM_3
    Then the difference between RAM_1 and RAM_3 is less than 1024 MB
```

Reply:
I like that, replaced the old steps with these in latest commit.